File: go/src/leetcode/offer/Offer50.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/31 23:54
 */

func firstUniqChar(s string) byte {
	cnt := [26]int{}
	for _, ch := range s {
		cnt[ch-'a']++
	}
	for i, ch := range s {
		if cnt[ch-'a'] == 1 {
			return s[i]
		}
	}
	return ' '
}

File: java/src/leetcode/offer/Offer50.java
package leetcode.offer;

import java.util.*;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/9/13 23:21
 */
public class Offer50 {

    public char firstUniqChar(String s) {
        Map<Character, Integer> frequency = new HashMap<>();
        for (int i = 0; i < s.length(); ++i) {
            char ch = s.charAt(i);
            frequency.put(ch, frequency.getOrDefault(ch, 0) + 1);
        }
        for (int i = 0; i < s.length(); ++i) {
            if (frequency.get(s.charAt(i)) == 1) {
                return s.charAt(i);
            }
        }
        return ' ';
    }
}
